ethereum/contracts/ETHApp.sol

```diff
@@ -82,7 +85,13 @@ contract ETHApp is RewardSource, AccessControl {
             "Invalid channel ID"
         );
 
-        emit Locked(msg.sender, _recipient, msg.value, _paraId);
+        require(
```

Comment:
I would remove this check, as I don't think it helps much. Even if the check passes, the XCM transfer could still fail further down the line if the fee isn't enough to buy the necessary weight. And there is no way for the contract to know those fee/weight numbers.

The user or its proxy (a client such as our web frontend) should take responsibility for  querying the destination parachains to determine how much the fee is, and only then issue a `lock` call to this contract.

Reply:
yeh - agree. It would be great if you can address this in the doc you're writing up with all the potential edge cases @alistair-singh and then we can work confirm whether or not it makes sense to check the fee here at all based on a more comprehensive review of all the edge cases

Reply:
Just to make sure we're all aligned here - Lets get PR merged in first (after addressing my last batch of comments), and then circle back with a doc & final PR to round out any rough edges regarding fees & XCM error-handling. And by XCM error-handling I mean the concepts introduced in this doc: https://medium.com/polkadot-network/xcm-part-iii-execution-and-error-management-ceb8155dd166, which mostly detail how to manage xcm script errors on the remote parachain.

I'm also going through the linear project at https://linear.app/snowfork/project/one-click-tofrom-any-parachain-49a0031f4a9b/SNO, cutting tickets to track the last remaining work.

Reply:
@vgeddes yep! just make sure to include all these above tasks around documentation, edge-case exploration and fixes/implementation as linear tasks before merging

Reply:
Done: https://linear.app/snowfork/project/one-click-to-any-parachain-49a0031f4a9b/SNO

Reply:
Removed checks.